cmd/antrea-agent/util_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetAvailableNodePortAddresses(t *testing.T) {
+	testCases := []struct {
+		name                        string
+		nodePortAddressesFromConfig []string
+		excludeDevices              []string
+	}{
+		{
+			name:                        "empty nodePortAddressesFromConfig",
+			nodePortAddressesFromConfig: []string{},
+			excludeDevices:              []string{"antrea-egress0", "antrea-ingress0", "kube-ipvs0", "antrea-gw0"},
+		},
+		{
+			name:                        "good path with nodePortAddressesFromConfig",
+			nodePortAddressesFromConfig: []string{"1.2.3.0/24", "1.2.3.4/32"},
+			excludeDevices:              []string{"antrea-egress0", "antrea-ingress0", "kube-ipvs0", "antrea-gw0"},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, _, err := getAvailableNodePortAddresses(tc.nodePortAddressesFromConfig, tc.excludeDevices)
+			assert.NoError(t, err)
+		})
+	}
+}
+
+func TestParsePortRange(t *testing.T) {
+	testCases := []struct {
+		name          string
+		portRangeStr  string
+		expectedstart int
+		expectedend   int
+		expectederr   string
+	}{
+		{
+			name:         "wrong port range format",
+			portRangeStr: "200 300",
+			expectederr:  fmt.Sprintf("wrong port range format: 200 300"),
+		},
+		{
+			name:         "length of port range string not equals to 2",
+			portRangeStr: "6110-7000-8000",
+			expectederr:  fmt.Sprintf("wrong port range format: 6110-7000-8000"),
+		},
+		{
+			name:         "wrong port range type value for start",
+			portRangeStr: "wrong-6200",
+			expectederr:  fmt.Sprintf("strconv.Atoi: parsing \"wrong\": invalid syntax"),
+		},
+		{
+			name:         "wrong port range type value for end",
+			portRangeStr: "6100-wrong",
+			expectederr:  fmt.Sprintf("strconv.Atoi: parsing \"wrong\": invalid syntax"),
+		},
+		{
+			name:          "valid range",
+			portRangeStr:  "61000-62000",
+			expectedstart: 61000,
+			expectedend:   62000,
+			expectederr:   "",
```

Comment:
you can just remove this line it will automatically be an empty string if no value is specified.